src/azure-cli/azure/cli/command_modules/acr/task.py

```diff
@@ -411,46 +417,59 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals, too-many-statements
     arguments = _get_all_override_arguments(arg, secret_arg)
     set_values = _get_all_override_arguments(set_value, set_secret)
 
-    if file and file.endswith(ALLOWED_TASK_FILE_TYPES):
-        step = FileTaskStepUpdateParameters(
-            task_file_path=file,
-            values_file_path=values,
-            context_path=context_path,
-            context_access_token=git_access_token,
-            values=set_values
-        )
-    elif file and not file.endswith(ALLOWED_TASK_FILE_TYPES):
-        step = DockerBuildStepUpdateParameters(
-            image_names=image_names,
-            is_push_enabled=not no_push,
-            no_cache=no_cache,
-            docker_file_path=file,
-            arguments=arguments,
+    if context_path is None:
+        yaml_template = get_yaml_template(
+            cmd_value, timeout, file)
+        import base64
+        step = EncodedTaskStepUpdateParameters(
+            encoded_task_content=base64.b64encode(
+                yaml_template.encode()).decode(),
             context_path=context_path,
             context_access_token=git_access_token,
-            target=target
+            values=(set_value if set_value else []) + (set_secret if set_secret else [])
         )
-    elif step:
-        if hasattr(step, 'docker_file_path'):
-            step = DockerBuildStepUpdateParameters(
-                image_names=image_names,
-                is_push_enabled=not no_push,
-                no_cache=no_cache,
-                docker_file_path=file,
-                arguments=arguments,
-                context_path=context_path,
-                context_access_token=git_access_token,
-                target=target
-            )
+    else:
+        if file:
```

Comment:
Question, if file is null here, do we need read file from existing task and see if it is null?